packages/ava/src/advisor/advise-pipeline/advicesForChart.ts

```diff
@@ -34,13 +34,23 @@ export function advicesForChart(
 
   // get dataProps from DataFrame
   let dataPropsForAdvice: BasicDataPropertyForAdvice[];
+  const defaultDataProps = dataFrame.info() as BasicDataPropertyForAdvice[];
   if (dataProps) {
     // filter out fields that are not included for advising
-    dataPropsForAdvice = fields
+    const inputDataProps = fields
       ? dataProps.filter((dataProp: BasicDataPropertyForAdvice) => fields.includes(dataProp.name))
       : dataProps;
+    dataPropsForAdvice = defaultDataProps.map((item) => {
```

Comment:
The original implementation would return an object with size of `fields` or size of `dataFrame.info()` , the former is less than or equal to the latter.

The size of the new return is always same with `dataFrame.info()`.

Are you sure this is the expected effect？

Reply:
You are right. My former implementation will not return an object with the size of `fields`. It has been revised.